src/vs/workbench/services/editor/browser/editorOverrideService.ts

```diff
@@ -40,18 +42,41 @@ export class EditorOverrideService extends Disposable implements IEditorOverride
 	readonly _serviceBrand: undefined;
 
 	private _contributionPoints: Map<string | glob.IRelativePattern, ContributionPoints> = new Map<string | glob.IRelativePattern, ContributionPoints>();
+	private static readonly overrideCacheStorageID = 'editorOverrideService.cache';
+	private cache: Set<string> | undefined;
 
 	constructor(
 		@IEditorGroupsService private readonly editorGroupService: IEditorGroupsService,
 		@IConfigurationService private readonly configurationService: IConfigurationService,
 		@IQuickInputService private readonly quickInputService: IQuickInputService,
 		@INotificationService private readonly notificationService: INotificationService,
 		@ITelemetryService private readonly telemetryService: ITelemetryService,
+		@IStorageService private readonly storageService: IStorageService,
+		@IExtensionService private readonly extensionService: IExtensionService
 	) {
 		super();
+		// Read in the cache on statup
+		this.cache = new Set<string>(JSON.parse(this.storageService.get(EditorOverrideService.overrideCacheStorageID, StorageScope.GLOBAL, JSON.stringify([]))));
+		this.storageService.remove(EditorOverrideService.overrideCacheStorageID, StorageScope.GLOBAL);
+
+		this._register(this.storageService.onWillSaveState(() => {
+			// We want to store the glob patterns we would activate on, this allows us to know if we need to await the ext host on startup for opening a resource
+			this.cacheContributionPoints();
```

Comment:
A slight improvement would be to only persist the contribution points into our cache if they have actually changed when last persisted in the session. As I said before `onWillSaveState` will be called often in web where we have no other way of persisting UI state on shutdown, so the code ideally is efficient.

Reply:
I'm not sure how we would know if the contribution states have changed. We know when items are inserted, but the removal lifecycle is managed by who ever owns the returned disposable.

Reply:
Yeah not a blocker, just wanted to bring this up. We call this method every 5s in web.